pkg/solana/fees/block_history_test.go

```diff
@@ -88,9 +90,197 @@ func TestBlockHistoryEstimator(t *testing.T) {
+		cfg.On("ComputeUnitPriceMin").Return(min).Once()
+		cfg.On("ComputeUnitPriceMax").Return(max).Once()
+		cfg.On("BlockHistoryPollPeriod").Return(pollPeriod).Once()
+		cfg.On("BlockHistoryDepth").Return(depth).Once()
+	}
+
+	initializeEstimator := func(t *testing.T, rwLoader *utils.LazyLoad[client.ReaderWriter], cfg *cfgmock.Config, lgr logger.Logger, ctx context.Context) *blockHistoryEstimator {
+		estimator, err := NewBlockHistoryEstimator(rwLoader, cfg, lgr, MultipleBlocksEstimator)
+		require.NoError(t, err, "Failed to create BlockHistoryEstimator")
+
+		require.NoError(t, estimator.Start(ctx), "Failed to start BlockHistoryEstimator")
+
+		// Ensure estimator is closed after the test
+		t.Cleanup(func() {
+			require.NoError(t, estimator.Close(), "Failed to close BlockHistoryEstimator")
+		})
+
+		return estimator
+	}
+
+	// Read multiple blocks from JSON file
+	testBlocksData, err := os.ReadFile("./multiple_blocks_data.json")
+	require.NoError(t, err)
+	var testBlocks []*rpc.GetBlockResult
+	require.NoError(t, json.Unmarshal(testBlocksData, &testBlocks))
+	require.GreaterOrEqual(t, len(testBlocks), int(blockHistoryDepth), "Not enough blocks in JSON to match blockHistoryDepth")
+
+	// Extract slots and compute unit prices from the blocks
+	// We'll consider the last 'blockHistoryDepth' blocks
+	var testSlots []uint64
+	var testPrices []ComputeUnitPrice
+	startIndex := len(testBlocks) - int(blockHistoryDepth)
+	testBlocks = testBlocks[startIndex:]
+	for _, block := range testBlocks {
+		// extract compute unit prices and get median from each block
+		slot := block.ParentSlot + 1
+		testSlots = append(testSlots, slot)
+		feeData, err := ParseBlock(block)
+		require.NoError(t, err, "Failed to parse block at slot %d", slot)
+		require.NotEmpty(t, feeData.Prices, "No compute unit prices found in block at slot %d", slot)
+		medianPrice, err := mathutil.Median(feeData.Prices...)
+		testPrices = append(testPrices, medianPrice)
+	}
+	testSlotsResult := rpc.BlocksResult(testSlots)
+
+	t.Run("Successful MultipleBlocksEstimator", func(t *testing.T) {
+		// Setup
+		rw := clientmock.NewReaderWriter(t)
+		rwLoader := utils.NewLazyLoad(func() (client.ReaderWriter, error) {
+			return rw, nil
+		})
+		cfg := cfgmock.NewConfig(t)
+		setupConfigMock(cfg, defaultPrice, min, max, pollPeriod, blockHistoryDepth)
+		rw.On("SlotHeight", mock.Anything).Return(testSlots[len(testSlots)-1], nil).Times(1)
+		rw.On("GetBlocksWithLimit", mock.Anything, mock.Anything, mock.Anything).
+			Return(&testSlotsResult, nil).Times(1)
+		for i, slot := range testSlots {
+			rw.On("GetBlock", mock.Anything, slot).
+				Return(testBlocks[i], nil).Times(1)
+		}
+		lgr, logs := logger.TestObserved(t, zapcore.DebugLevel)
+		estimator := initializeEstimator(t, rwLoader, cfg, lgr, tests.Context(t))
+
+		// Calculate expected median price from all the blocks and check estimated price
+		tests.AssertLogEventually(t, logs, "BlockHistoryEstimator: updated")
```

Comment:
I know we use this in other parts of our code base but I think we want to minimize our use of eventually. Think it leads to potential flakiness. Could we instead make `calculatePrice` price a public method and call that directly in the test? Then we can be sure the price is set and check `BaseComputeUnitPrice` after.